src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -1032,48 +1032,116 @@ export class BrowsingContextImpl {
     return await this.#locateNodesByLocator(this.#defaultRealm, params.locator);
   }
 
-  #getLocatorFunction(locator: BrowsingContext.Locator): string {
+  #getLocatorDelegate(locator: BrowsingContext.Locator): {
+    functionDeclaration: string;
+    argumentsLocalValues: Script.LocalValue[];
+  } {
     switch (locator.type) {
       case 'css':
-        return String((cssSelector: string) => {
-          const results = document.querySelectorAll(cssSelector);
-          const array = [];
-          for (const item of results) {
-            array.push(item);
-          }
-          return array;
-        });
+        return {
+          functionDeclaration: String((cssSelector: string) => {
+            const results = document.querySelectorAll(cssSelector);
+            const array = [];
+            for (const item of results) {
+              array.push(item);
+            }
+            return array;
```

Comment:
Array.from(document.querySelectorAll(cssSelector))?

Reply:
Out of scope of this PR? I will rework it anyways to adopt other options.

Reply:
sg